cumulus/client/cirrus-executor/src/aux_schema.rs

```diff
@@ -50,18 +50,18 @@ pub(super) fn write_execution_receipt<Backend: AuxStore, Block: BlockT, PBlock:
 
 	let mut new_first_saved_receipt = first_saved_receipt;
 
-	let keys_to_delete = if best_execution_chain_number.saturating_sub(first_saved_receipt) >=
-		PRUNING_DEPTH.into()
+	let mut keys_to_delete = vec![];
+
+	if let Some(delete_receipts_to) = best_execution_chain_number
+		.saturated_into::<BlockNumber>()
+		.checked_sub(PRUNING_DEPTH)
 	{
-		// `first_saved_receipt` starts from 1 instead of 0, hence `PRUNING_DEPTH` - 1.
 		new_first_saved_receipt =
-			best_execution_chain_number.saturating_sub((PRUNING_DEPTH - 1).saturated_into());
-
-		let mut keys_to_delete = vec![];
-		let mut to_delete_start = first_saved_receipt;
-		while to_delete_start < new_first_saved_receipt {
+			delete_receipts_to.saturated_into::<NumberFor<Block>>() + One::one();
```

Comment:
`into` can be used here as it's the concrete type u32 to the generic BlockNumber.

Reply:
Yes, it just wasn't able to infer types automatically, just pushed the change with suggestion